### ISSUE TITLE
feat: new PR check for PR title format

### ISSUE THREAD

**Issue by apeabody** [GoogleCloudPlatform/cloud-foundation-toolkit#1304]
We should add a new PR check to validate that PR titles follow the www.conventionalcommits.org format for release-please.

**Comment by bharathkkb** [comment]
Good idea! We could use write a  [github script](https://github.com/actions/github-script) on pull_request events to check if title is correctly formatted.

**Comment by bharathkkb** [comment]
I also saw https://github.com/googleapis/repo-automation-bots/tree/main/packages/conventional-commit-lint

**Comment by daniel-cit** [comment]
When we have a single commit in the PR the PR title and the single commit message should be the same. 
when we have multiple commits, only the PR title is validated.

The error message for the single commit case could explicitly say that in this case both the PR title and the single commit should follow the convention.

https://github.com/googleapis/repo-automation-bots/blob/main/packages/conventional-commit-lint/__snapshots__/conventional-commit-lint.js#L53-L62

```js
exports['ConventionalCommitLint adds a comment when the commit message and the PR title differ 1'] = {
  "head_sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
  "conclusion": "failure",
  "name": "conventionalcommits.org",
  "output": {
    "title": "Commit message did not follow Conventional Commits",
    "summary": "Some of your commit messages failed linting.\n\nVisit [conventionalcommits.org](https://conventionalcommits.org) to learn our conventions.\n\nRun `git commit --amend` and edit your message to match Conventional Commit guidelines.",
    "text": ":x: The following linting errors found:\n* subject may not be empty\n* type may not be empty\nfor the following input:\n\"*fix all the bugs*\"\n\n"
  }
}
```

 It is in the key `'ConventionalCommitLint adds a comment when the commit message and the PR title differ 1'` but not in the body.

it happens in this PR https://github.com/terraform-google-modules/terraform-example-foundation/pull/907/checks

**Comment by apeabody** [comment]
OK, so this is the section the check summary message: https://github.com/googleapis/repo-automation-bots/blob/main/packages/conventional-commit-lint/src/utils.ts#L182

We could probably set a flag in this section that is used to alter customize the message: https://github.com/googleapis/repo-automation-bots/blob/main/packages/conventional-commit-lint/src/utils.ts#L125

**Comment by tmatsuo** [comment]
> The error message for the single commit case could explicitly say that in this case both the PR title and the single commit should follow the convention.

When the PR title differ from the commit message, we assume someone edit the PR title to please the bot.

Unfortunately the bot needs to check the commit message when there's only one commit because Github will use the commit message as the final commit message on the main branch.

There are two exceptions though.
1. When there's a label 'automerge'. This label is for one of other bots of ours for automerging PRs. This bot will always use the PR title for the final commit message, so the conventional commit bot can safely assume the PR title will be used.
2. When the PR author use Github's automerge feature with squash commit. In this case, Github will also use the PR title as the final commit message, so the conventional commit bot can also check against the PR title.

I suggest you try Github's automerge feature and let us know what you think.

**Comment by tmatsuo** [comment]
Well, actually, I'm going to introduce a new config to the bot:
https://github.com/googleapis/repo-automation-bots/pull/4770

**Comment by apeabody** [comment]
Thanks @tmatsuo!
